Adyen/CheckoutConfiguration/CheckoutBaseCallbacks.swift

```diff
@@ -27,3 +27,49 @@ package protocol CheckoutBaseCallbacks {
     
     var onComplete: CheckoutSuccessHandler? { get set }
 }
+
+/// Data model to contain relevant parts of `/payments` and `/payment/details` call responses.
+public struct CheckoutPaymentsResponse: Decodable, Sendable {
+    
+    // TODO: should it be string or enum?
+    public let resultCode: String
+    
+    public let action: Action?
+    
+    public let order: PartialPaymentOrder?
+    
+    public init(from decoder: any Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.resultCode = try container.decode(String.self, forKey: .resultCode)
+        self.action = try container.decodeIfPresent(Action.self, forKey: .action)
+        self.order = try container.decodeIfPresent(PartialPaymentOrder.self, forKey: .order)
+    }
+    
+    private enum CodingKeys: CodingKey {
+        case resultCode
+        case action
+        case order
+    }
+}
+
+/// Data model that contains information regarding the status of a payment.
+public struct CheckoutResult {
+    
+    // TODO: string or enum?
+    public let resultCode: String
```

Comment:
It's a string so let's make it enum, this prevents mistypes when comparing and makes it easier to update in the future.